src/sets.jl

```diff
@@ -669,18 +669,69 @@ end
 
 Base.:(==)(set1::IndicatorSet{A, S}, set2::IndicatorSet{A, S}) where {A, S} = set1.set == set2.set
 
+"""
+    Complements(dimension::Int)
+
+The set corresponding to a mixed complementarity constraint.
+
+Complementarity constraints should be specified with an
+[`AbstractVectorFunction`](@ref)-in-`Complements(dimension)` constraint.
+
+The dimension of the vector-valued function `F` must be `2 * dimension`. This
+defines a complementarity constraint between the scalar function `F[i]` and the
+variable in `F[i + dimension]`. Thus, `F[i + dimension]` must be interpretable
+as a single variable `x_i` (e.g., `1.0 * x + 0.0`).
+
+If the variable `x_i` is constrained in `Interval(lb_i, ub_i)`, then
+mathematically, the mixed complementarity problem is to find a solution such
+that at least one of the following holds:
+
```

Comment:
I would rephase the second part as follows:
"the mixed complementarity problem consists in finding `x_i` in the interval `[lb_i, ub_i]` such that the following holds:"

All statements must hold. I've discussed this a bit further later.